aptos-move/move-examples/defi/aptoswap/core/sources/liquidity_pool.move

```diff
@@ -0,0 +1,417 @@
+module pool_manager::liquidity_pool {
+    use aptos_framework::fungible_asset::{Metadata, FungibleStore, FungibleAsset, deposit};
+    use aptos_framework::object::{Object, ExtendRef, generate_signer_for_extending, generate_extend_ref};
+    use aptos_framework::fungible_asset;
+    use aptos_framework::object;
+    use aptos_framework::primary_fungible_store;
+    use aptos_std::comparator;
+    use aptos_std::math128;
+    use aptos_std::math64;
+    use aptos_std::smart_table::SmartTable;
+    use aptos_std::smart_table;
+    use std::bcs;
+    use std::error;
+    use std::option;
+    use std::string;
+    use std::vector;
+
+    use pool_manager::utilities;
+
+    const MINIMAL_LIQUIDITY: u64 = 1000;
+
+    // fee is basis point based so the scale is 10000.
+    const FEE_SCALE: u64 = 10000;
+
+    const ERR_SWAP_PAIR_CANNOT_BE_THE_SAME: u64 = 1;
+    const ERR_POOL_EXISTENCE: u64 = 2;
+    const ERR_NOT_ENOUGH_LIQUIDITY: u64 = 3;
+    const ERR_NOT_ENOUGH_LP_TOKEN_TO_BURN: u64 = 4;
+    const ERR_ZERO_COIN_TO_SWAP: u64 = 5;
+    const ERR_INCORRECT_SWAP_DUE_TO_REDUCED_LIQUIDITY: u64 = 6;
+
+    /// Stored at lp token metadata object to find the liquidity pool.
+    /// x address must be lexicographically smaller than y address.
```

Comment:
Is this still needed for fungible assets outside of Solidity and CoinType?

Reply:
(a, b) and (b, a) should share a unique id.

Reply:
But why?